### PR TITLE
[Saleable] Add interface to Artwork & EditionSet.

### PR DESCRIPTION
**Pull request by alloy**
Yay, our first real interface!

I’m not sure why the `EditionSet` type is called like that, isn’t each an entry in the set? Anyways, after thinking about `Edition`, `Piece`, etc for a while, I figured I’d go with an adjective form for interfaces as can be found in various examples, e.g. [`Searchable`](https://medium.com/the-graphqlhub/graphql-tour-interfaces-and-unions-7dd5be35de0d). Obviously I then had to choose between the various flavours of English (sellable, saleable, scalable) and settled on the one that seemed the most used in both English and Simplified (US) English.

----

This now allows you to have a fragment on `Saleable` and query any type that implements it:

```graphql
{
  artwork(id: "banksy-flower-bomber-by-brandalism") {
    ...Piece
    edition_sets {
      ...Piece
    }
  }
}

fragment Piece on Saleable {
  dimensions {
    in
  }
  edition_of
}
```

Meaning that containers that fetch this data can now be generic:

```tsx
export class SizeInfo extends React.Component<SizeInfoProps> {
  render() {
    const {
      piece: { dimensions, edition_of },
    } = this.props
    const dimensionsDescription = dimensions &&
            (dimensions.in || dimensions.cm) &&
            [dimensions.in, dimensions.cm].join("; ")
    return (
      <SizeInfoContainer color="black60" textAlign="left">
        <Serif size="2">
          {dimensionsDescription}
        </Serif>
        {edition_of && <Serif size="2">{edition_of}</Serif>}
      </SizeInfoContainer>
    )
  }
}

export const SizeInfoFragmentContainer = createFragmentContainer(
  SizeInfo,
  graphql`
    fragment SizeInfo_piece on Saleable {
      dimensions {
        in
        cm
      }
      edition_of
    }
  `
)
```

…and be instantiated with any type that implements the interface:

```tsx
<SizeInfo piece={this.props.artwork} />
<SizeInfo piece={this.props.artwork.edition_sets[0]} />
```

### CODE REVIEW

File: src/schema/saleable.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import Dimensions from "./dimensions"
+import { GraphQLInterfaceType, GraphQLBoolean, GraphQLString } from "graphql"
+
+export const Saleable = new GraphQLInterfaceType({
```
Comment (mzikherman) [inline src/schema/saleable.ts]: `Sellable` might be a better name than `Saleable`?

**Comment by mzikherman** [comment]
Other than the naming Saleable vs Sellable, LGTM!

File: src/schema/saleable.ts (hunk 1 above)
Comment (alloy) [reply src/schema/saleable.ts]: I went down a little 🐰 🕳 for this one. E.g. from https://www.englishforums.com/English/SellableVersusSaleableVesus-Salable/nrchc/post.htm:

> I disagree. I have two business degrees, minored in linguistics, and have been working in the retail operations and marketing field all my adult life. Salable, saleable, and sellable are often pronounced the same and really all mean the same thing-- able to be sold. There is no connotation or implication of profit margin. With that being said, however, in the US "saleable" is BY FAR the most commonly used in writing. When I see that someone has written "salable," I pause to wonder where they might be from. When I see that someone has written "sellable," I know they likely aren't a business pro.

There’s more on e.g. https://forum.wordreference.com/threads/saleable-vs-sellable.523745/

**Comment by orta** [comment]
Yeah, looks right to me, man this stuff is verbose, OK, I'll try and make a metaphysics + SDL PR this week.

**Comment by joeyAghion** [comment]
I agree (at first glance, and not having gone down the rabbit hole) about "sellable" being more intuitive for this purpose. In any case I like naming this according to its behavior rather than trying to find a general enough noun to cover all uses. This is more naturally composable.

Regarding why an "edition set" is called that, each `EditionSet` record represents 1 _or more_ physical objects sharing metadata like dimension and price. While each represents works that are presumably part of the same "limited edition," it's not precisely the same as that "edition," because it might not include all instances (e.g., it might only include those numbered "1-3, and 9" of an edition of 10) or price data might need to vary across the actual "edition" (e.g., those numbered 1-2 might be offered at a higher price than 3-10, so be represented by distinct `EditionSet`s).

**Comment by alloy** [comment]
I can still change the naming if multiple people prefer Sellable, I have no personal preference and it’s not being used in production yet.

Thanks for the explanation re edition sets :pray:

**Comment by alloy** [comment]
https://github.com/artsy/metaphysics/pull/1142

**Pull request merged**